stats/opentelemetry/e2e_test.go

```diff
@@ -36,6 +36,21 @@ import (
 	"go.opentelemetry.io/otel/sdk/metric/metricdata/metricdatatest"
 )
 
+func ExampleNewMetrics() {
+	// Will disable metrics, set with no metrics specified.
+	NewMetrics()
+	// Will enable these two metrics.
+	NewMetrics(ClientAttemptDuration, ServerCallDuration)
+	// Use DefaultClientMetrics to enable default client metrics. Equivalent to
+	// unset, which will pick up defaults.
+}
+
+func ExampleMetric_Remove() {
+	// Use DefaultClientMetrics without ClientAttemptDuration and
+	// ClientCallDurationName.
+	DefaultClientMetrics.Remove(ClientAttemptDuration, ClientCallDurationName)
+}
```

Comment:
I think these would be more useful on the DialOption / ServerOption, or Options, and should show how to build the DialOption.  Just one line of code isn't super useful.

E.g. something like:

```go
func ExampleDialOption_ExcludeMetrics() {
	// To exclude specific metrics, initialize Options as follows:
	otelOpts := opentelemetry.Option{
		MetricsOptions: MetricsOptions{
			Metrics: DefaultClientMetrics.Remove(opentelemetry.ClientAttemptDuration,
		},
	}
	dialOption := opentelemetry.DialOption(otelOpts)
	cc, _ := grpc.Dial("<target string>",  dialOption)
	// Handle err.
	defer cc.Close()
}
```

Reply:
Don't the examples technically have to be linked to a single exported function from the package?

Reply:
No, there are many ways of attaching examples at different places in the package.  Check the testing documentation for more details: https://pkg.go.dev/testing#hdr-Examples

Reply:
Added numerous examples.